core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -694,6 +694,53 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     assertEquals(outOfRangePos.toLong, outOfRangePartitions.get(tp))
   }
 
+  @Test
+  def testFetchOutOfRangeOffsetResetConfigEarliest(): Unit = {
+    this.consumerConfig.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+    // ensure no records arrive before poll is called so that the offset actually gets reset
+    this.consumerConfig.setProperty(ConsumerConfig.FETCH_MAX_WAIT_MS_CONFIG, "0")
```

Comment:
I think this comment is not clear. Maybe 
// ensure no in-flight fetch request so that the offset can be reset immediately